receiver/googlecloudpubsubreceiver/factory.go

```diff
@@ -107,7 +107,7 @@ func (factory *pubsubReceiverFactory) CreateLogs(
 	cfg component.Config,
 	consumer consumer.Logs,
 ) (receiver.Logs, error) {
-	err := cfg.(*Config).validateForLog()
+	err := cfg.(*Config).validate()
```

Comment:
The validate function doesn't have the same checks as `validateForLogs`.  Does it need updated?

Reply:
The commit states the reason. The logic is moved toward the start(), including extra tests.